src/Microsoft.Health.Fhir.Shared.Core/Extensions/ReindexMediatorExtensions.cs

```diff
@@ -0,0 +1,59 @@
+﻿// -------------------------------------------------------------------------------------------------
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License (MIT). See LICENSE in the repo root for license information.
+// -------------------------------------------------------------------------------------------------
+
+using System;
+using System.Threading;
+using System.Threading.Tasks;
+using EnsureThat;
+using MediatR;
+using Microsoft.Health.Fhir.Core.Messages.Reindex;
+using Microsoft.Health.Fhir.Core.Models;
+
+namespace Microsoft.Health.Fhir.Core.Extensions
+{
+    public static class ReindexMediatorExtensions
+    {
+        public static async Task<ResourceElement> CreateReindexJobAsync(
+            this IMediator mediator,
+            int? maximumConcurrency,
+            string scope,
+            CancellationToken cancellationToken)
+        {
+            EnsureArg.IsNotNull(mediator, nameof(mediator));
+
+            var request = new CreateReindexRequest(maximumConcurrency, scope);
+
+            CreateReindexResponse response = await mediator.Send(request, cancellationToken);
+            return response.Job.ToParametersResourceElement();
+        }
+
+        public static async Task<ResourceElement> GetReindexJobAsync(this IMediator mediator, string jobId, CancellationToken cancellationToken)
+        {
+            EnsureArg.IsNotNull(mediator, nameof(mediator));
+
```

Comment:
Should we ensure jobId here? EnsureArg.IsNotNullOrWhiteSpace(jobId, nameof(jobId))

Reply:
Currently the handler checks for jobId being null, and if so will return a list of reindex jobs.  But perhaps this should be changed to two different handlers.